cardinal/ecs/engine.go

```diff
@@ -467,32 +471,36 @@ func (e *Engine) Tick(ctx context.Context) error {
 		}
 	}
 	e.timestamp.Store(uint64(startTime.Unix()))
-	allSystemsStartTime := time.Now()
 	for i, sys := range e.systems {
 		nameOfCurrentRunningSystem = e.systemNames[i]
 		eCtx := NewEngineContextForTick(e, txQueue, e.systemLoggers[i])
 		systemStartTime := time.Now()
 		err := eris.Wrapf(sys(eCtx), "system %s generated an error", nameOfCurrentRunningSystem)
 		statsd.EmitTickStat(systemStartTime, nameOfCurrentRunningSystem)
-		nameOfCurrentRunningSystem = nullSystemName
+		nameOfCurrentRunningSystem = unnamedSystem
 		if err != nil {
 			return err
 		}
 	}
-	statsd.EmitTickStat(allSystemsStartTime, "all_systems")
+	statsd.EmitTickStat(startTime, "all_systems")
 	if e.eventHub != nil {
-		flushEventHubStartTime := time.Now()
 		// engine can be optionally loaded with or without an eventHub. If there is one, on every tick it must flush events.
 		e.eventHub.FlushEvents()
-		statsd.EmitTickStat(flushEventHubStartTime, "flush_events")
+		statsd.EmitTickStat(time.Now(), "flush_events")
```

Comment:
EmitTickStat takes the difference between the given time and time.Now and emits the delta as a metric. This is just going to cause every "flush_events" metric to be 0. Same with "finalize" below.

Reply:
ah okay, will revert. linter was complaining this function is too long. ill find another way to fix